File: scripts/debug-pdf.mjs
import { chromium } from 'playwright'

const browser = await chromium.launch({ headless: true })
const page = await browser.newPage({ viewport: { width: 1400, height: 900 } })

const logs = []
page.on('console', (msg) => {
  const text = msg.text()
  logs.push(`${msg.type()}: ${text}`)
})
page.on('pageerror', (err) => {
  logs.push(`pageerror: ${err.message}`)
})

await page.goto('http://localhost:3000')
await page.waitForTimeout(3000)

// Screenshot of the full page
await page.screenshot({ path: '/Users/louiss/Programming/Projects/portfolio-website/scripts/debug-pdf-full.png', fullPage: true })

// Get the PDF viewer element bounding box and dimensions
const pdfViewer = await page.$('[data-slot="pdf-viewer"]')
if (pdfViewer) {
  const box = await pdfViewer.boundingBox()
  console.log('PDF viewer bounding box:', JSON.stringify(box))
  const height = await pdfViewer.evaluate((el) => el.clientHeight)
  const width = await pdfViewer.evaluate((el) => el.clientWidth)
  console.log('PDF viewer client size:', { width, height })
}

const scrollViewport = await page.$('[data-slot="scroll-area-viewport"]')
if (scrollViewport) {
  const box = await scrollViewport.boundingBox()
  console.log('Scroll viewport bounding box:', JSON.stringify(box))
  const height = await scrollViewport.evaluate((el) => el.clientHeight)
  const width = await scrollViewport.evaluate((el) => el.clientWidth)
  console.log('Scroll viewport client size:', { width, height })
}

const pdfContent = await page.$('[data-slot="pdf-viewer-scroll-content"]')
if (pdfContent) {
  const box = await pdfContent.boundingBox()
  console.log('PDF scroll content bounding box:', JSON.stringify(box))
}

console.log('--- Console logs ---')
console.log(logs.join('\n'))

await browser.close()

File: scripts/debug_pdf.py
from playwright.sync_api import sync_playwright
import json

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page(viewport={"width": 1400, "height": 900})

    logs = []
    def on_console(msg):
        logs.append(f"{msg.type}: {msg.text}")
    page.on("console", on_console)
    page.on("pageerror", lambda err: logs.append(f"pageerror: {err.message}"))

    page.goto("http://localhost:3000")
    page.wait_for_timeout(8000)

    page.screenshot(path="/Users/louiss/Programming/Projects/portfolio-website/scripts/debug-pdf-full.png", full_page=True)

    pdf_viewer = page.query_selector('[data-slot="pdf-viewer"]')
    if pdf_viewer:
        box = pdf_viewer.bounding_box()
        print("PDF viewer bounding box:", json.dumps(box))
        size = pdf_viewer.evaluate("el => ({ w: el.clientWidth, h: el.clientHeight })")
        print("PDF viewer client size:", size)
    else:
        print("PDF viewer not found")

    scroll_viewport = page.query_selector('[data-slot="scroll-area-viewport"]')
    if scroll_viewport:
        box = scroll_viewport.bounding_box()
        print("Scroll viewport bounding box:", json.dumps(box))
        size = scroll_viewport.evaluate("el => ({ w: el.clientWidth, h: el.clientHeight })")
        print("Scroll viewport client size:", size)
    else:
        print("Scroll viewport not found")

    scroll_content = page.query_selector('[data-slot="pdf-viewer-scroll-content"]')
    if scroll_content:
        box = scroll_content.bounding_box()
        print("PDF scroll content bounding box:", json.dumps(box))
    else:
        print("PDF scroll content not found")

    # Count canvas and page elements
    page_count = page.evaluate("() => document.querySelectorAll('[data-pdf-viewer-page]').length")
    canvas_count = page.evaluate("() => document.querySelectorAll('canvas').length")
    print("PDF page elements:", page_count)
    print("Canvas elements:", canvas_count)

    print("--- Console logs ---")
    print("\n".join(logs))

    browser.close()
